Sources/SwiftDocC/Infrastructure/Bundle Assets/DataAssetManager.swift

```diff
@@ -172,18 +177,32 @@ public struct DataAsset: Codable {
     /// depending on the system's appearance.
     public var variants = [DataTraitCollection: URL]()
     
+    /// The metadata associated with each variant.
+    public var metadata = [URL : Metadata]()
```

Comment:
A data modeling question: What's the benefit of having metadata per variant instead of per asset? Would different variants of an SVG ever have different IDs?

Reply:
That's exactly the benefit: The ID is internal to the SVG and I think generally unknown to the developer/author so we can't depend on having the same ID per variant. I can imagine that in most cases the ID will be the same across variants but we can't depend on that without the potential of running into confusing breakage down the road.

Future metadata (maybe image size for a PNG?) - would also be variant (file) specific, not asset specific.